### ISSUE TITLE
transaction traces

### ISSUE THREAD

**Issue by dkatzan** [toncenter/ton-indexer#73]
Hi, I have some questions regarding traces, which seem to be a bit removed from the APIs, why is that?

1. Why is the [`get_transaction_trace`](https://github.com/toncenter/ton-indexer/blob/051eeeec73eab72dc808489288d2d1780498774a/indexer/indexer/api/api_v1/main.py#L350C11-L350C32) API hidden from schema and has a comment this is going to be removed so not to relay on it for production code?

2. What about the [`get_traces`](https://github.com/toncenter/ton-indexer/blob/051eeeec73eab72dc808489288d2d1780498774a/indexer/indexer/api/api_v1/main.py#L330)API? I see that is also hidden, can it be used?
I tried calling it with a specific transaction hash and got back an empty response

3. Why is the [`trace_id`](https://github.com/toncenter/ton-indexer/blob/051eeeec73eab72dc808489288d2d1780498774a/indexer/indexer/api/api_v1/schemas.py#L343) field from the transactions schema being commented out?

we would be very happy to use such endpoints in the indexer,
thx a lot for this project!

**Comment by kdimentionaltree** [comment]
Hello, for now the traces, events and actions API is in development. In the repo you may find the script [run_event_detector.sh](https://github.com/toncenter/ton-indexer/blob/master/run_event_detector.sh) which was a in-dev version of trace assembler.

After a while we integrated the TraceAssembler in [the C++ submodule](https://github.com/toncenter/ton-index-worker/blob/main/tondb-scanner/src/TraceAssembler.h) and changed database schema. Soon we will release this API in testnet.toncenter.com/api/v3 and publish the code of Golang-version of API.

**Comment by dkatzan** [comment]
Thx for u response,
is the run_even_detector script meant to be run in the background and constantly detecting and building new traces? allowing later to query for traces cheaply?

is there any expectations on releasing the v3 API for testnet? and for mainnet?

**Comment by kdimentionaltree** [comment]
Correct, "run_event_detector" is an early version of the system, which markups traces during the indexing process. It was integrated in ton-index-worker as [TraceAssembler](https://github.com/toncenter/ton-index-worker/blob/main/tondb-scanner/src/TraceAssembler.h).

ETA for "testnet" is the end of August (see [events section](https://testnet.toncenter.com/api/v3/index.html#/events)).

**Comment by abdallahabdellatif** [comment]
Hey @kdimentionaltree 
I can see trace_id is live on testnet now, is there an ETA for it being live on mainnet?

Recently I saw it was returned by transaction endpoints for mainnet but today I see it's revoked

Would appreciate updates on this and keep up the great work !
